experimental/devices/nrzled/nrz_spi.go

```diff
@@ -0,0 +1,453 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package nrzled
+
+import (
+	"errors"
+	"fmt"
+	"image"
+	"image/color"
+	"image/draw"
+
+	"periph.io/x/periph/conn/display"
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+)
+
+// SPIDev represents a strip of WS2812b LEDs as a strip connected over a SPI port.
+// It accepts a stream of raw RGB pixels and converts it to a bit pattern consistent
+// with the WS812b protocol.
+// Includes intensity and temperature correction.
+type SPIDev struct {
+	s         spi.Conn        //
+	numPixels int             //
+	rawBuf    []byte          // Raw buffer sent over SPI. Cached to reduce heap fragmentation.
+	pixels    []byte          // Double buffer of pixels, to enable partial painting via Draw(). Effectively points inside rawBuf.
+	rect      image.Rectangle // Device bounds
+	pixelMap  map[image.Point]int
+}
+
+// NewSPI returns a strip that communicates over SPI to NRZ encoded LEDs.
+//
+// Due to the tight timing demands of these LEDs,
+// the SPI port speed must be a reliable 2.5MHz
```

Comment:
Shouldn't it be 2.4MHz? I guess it doesn't work because of it being a multiple of the core's speed (?)
I know that the SPI port is rarely what is exactly asked. :/